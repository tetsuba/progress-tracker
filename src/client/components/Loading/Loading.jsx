import React from 'react'
import { Fade, Spinner } from 'react-bootstrap'

export default function Loading({ fade }) {
  const style = {
    width: '100%',
    height: '100%',
    position: 'absolute',
    top: '0',
    left: '0',
    zIndex: '1000',
    backgroundColor: 'rgba(0,0,0,0.5)',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  }
  return (
    <Fade in={fade}>
      <div style={style}>
        <Spinner animation="border" role="status" variant="light">
          <span className="sr-only">Loading...</span>
        </Spinner>
      </div>
    </Fade>
  )
}
