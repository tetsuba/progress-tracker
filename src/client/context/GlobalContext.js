import * as React from 'react'
// $FlowFixMe
import ModalProvider from './ModalContext'
// $FlowFixMe
import AuthenticatedProvider from './AuthenticatedContext'
import LoadingProvider from './LoadingContext'

type Props = {
  children: React.Node,
}

export default class GlobalContext extends React.Component<Props> {
  render() {
    return (
      <LoadingProvider>
        <ModalProvider>
          <AuthenticatedProvider>{this.props.children}</AuthenticatedProvider>
        </ModalProvider>
      </LoadingProvider>
    )
  }
}
