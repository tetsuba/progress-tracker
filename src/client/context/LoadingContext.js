import * as React from 'react'

export const LoadingContext = React.createContext({
  loading: false,
  showLoading: () => {},
  hideLoading: () => {},
})

type Props = {
  children?: React.Node,
}

export default function LoadingProvider(props: Props) {
  const { children } = props
  const [loading, setLoading] = React.useState(false)

  return (
    <LoadingContext.Provider
      value={{
        loading,
        showLoading: () => setLoading(true),
        hideLoading: () => setTimeout(() => setLoading(false), 300),
      }}
    >
      {children}
    </LoadingContext.Provider>
  )
}
