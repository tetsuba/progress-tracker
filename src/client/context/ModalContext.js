import * as React from 'react'

export const ModalContext = React.createContext({
  ModalTemplate: null,
  show: false,
  hideModal: () => {},
  toggleModal: () => {},
  addTemplate: () => {},
})

type Props = {
  children: React.Node,
}

export default class ModalProvider extends React.Component<Props> {
  state = {
    show: false,
    ModalTemplate: null,
    hideModal: () => this.setState({ show: false }),
    toggleModal: () =>
      this.setState({
        show: !this.state.show,
      }),

    addTemplate: (template) =>
      this.setState({
        ModalTemplate: template,
      }),
  }

  render() {
    return (
      <ModalContext.Provider value={this.state}>
        {this.props.children}
      </ModalContext.Provider>
    )
  }
}
