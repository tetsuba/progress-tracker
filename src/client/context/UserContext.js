import React, { Component } from 'react';

export const UserContext = React.createContext({
    userId: '',
    firstName: '',
    lastName: '',
    email: '',
    addUserId: () => {},
    userLoggedIn: () => {},
    userUpdate: () => {},
});


export default class UserCxt extends Component {
    state = {
        userId: '',
        firstName: '',
        lastName: '',
        email: '',
    };

    addUserId = (id) => this.setState({ userId: id, loggedIn: true });
    userLoggedIn = (data) => this.setState({
        firstName: data.firstName,
        lastName: data.lastName,
        email: data.email,
        userId: data.id,
    });
    userUpdate = (data) => this.setState({
        ...this.state,
        firstName: data.firstName,
        lastName: data.lastName,
    });

    render() {
        return (
            <UserContext.Provider value={{
                ...this.state,
                addUserId: this.addUserId,
                userLoggedIn: this.userLoggedIn,
                userUpdate: this.userUpdate,
            }}>
                { this.props.children }
            </UserContext.Provider>
        )
    }
}

