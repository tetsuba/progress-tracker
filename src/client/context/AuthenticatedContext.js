import React, { useState } from 'react'
import { useLocalStorage } from '../hooks/hooks'
import { TOKEN_KEY } from '../const/localStorage'
import { useQuery } from '@apollo/react-hooks'
import { IS_USER_SESSION_EXPIRED } from '../api/user/user.query'
import Loading from '../components/Loading/Loading'

export const AuthenticatedContext = React.createContext({
  authenticated: false,
  toggle: () => {},
})

export default function AuthenticatedCxt({ children }) {
  const [authenticated, setAuthenticated] = useState(false)
  const [setLocalStorage, token] = useLocalStorage(TOKEN_KEY)
  const { loading, data } = useQuery(IS_USER_SESSION_EXPIRED)

  /* Set authentication to true:
   * - data exists
   * - authenticated is false
   * - token exists
   *  */
  if (data && !authenticated && token) {
    setAuthenticated(true)
  }

  if (loading) return <Loading />

  return (
    <AuthenticatedContext.Provider
      value={{
        authenticated,
        toggle: (token) => {
          if (token) {
            setLocalStorage(token)
            setAuthenticated(true)
            console.log('+++ Login')
          } else {
            setLocalStorage('')
            setAuthenticated(false)
            console.log('--- Logout')
          }
        },
      }}
    >
      {children}
    </AuthenticatedContext.Provider>
  )
}
