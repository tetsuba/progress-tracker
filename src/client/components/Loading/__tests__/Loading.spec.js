import { mount } from 'enzyme'
import Loading from '../Loading'
import * as React from 'react'
import { LoadingContext } from '../../../context/LoadingContext'

describe('<Loading>', () => {
  it('should render loading screen', () => {
    const wrapper = mount(
      <LoadingContext.Provider value={{ loading: true }}>
        <Loading />
      </LoadingContext.Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should render nothing', () => {
    const wrapper = mount(
      <LoadingContext.Provider value={{ loading: false }}>
        <Loading />
      </LoadingContext.Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
