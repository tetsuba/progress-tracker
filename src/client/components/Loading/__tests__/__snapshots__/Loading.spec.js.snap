// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading> should render loading screen 1`] = `
<Loading>
  <Fade
    appear={false}
    in={true}
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <Transition
      addEndListener={[Function]}
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={300}
      unmountOnExit={false}
    >
      <div
        className="fade show"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.5)",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "left": "0",
            "position": "absolute",
            "top": "0",
            "width": "100%",
            "zIndex": "1000",
          }
        }
      >
        <Spinner
          animation="border"
          role="status"
          variant="light"
        >
          <div
            className="spinner-border text-light"
            role="status"
          >
            <span
              className="sr-only"
            >
              Loading...
            </span>
          </div>
        </Spinner>
      </div>
    </Transition>
  </Fade>
</Loading>
`;

exports[`<Loading> should render nothing 1`] = `<Loading />`;
